tls/s2n_security_policies.c

```diff
@@ -611,6 +612,12 @@ int s2n_config_set_cipher_preferences(struct s2n_config *config, const char *ver
     notnull_check(&config->security_policy->kem_preferences);
     notnull_check(&config->security_policy->signature_preferences);
     notnull_check(&config->security_policy->ecc_preferences);
+
+    if (s2n_is_in_fips_mode()) {
+        ENSURE_POSIX(config->security_policy->kem_preferences == &kem_preferences_null,
+                S2N_ERR_PQ_KEMS_DISALLOWED_IN_FIPS);
+    }
+
```

Comment:
This will return an error, but `GUARD(s2n_find_security_policy_from_version(version, &config->security_policy));` will have already set the security policy in the config. 

I think we need to pass a `temp_sec_policy` variable to `s2n_find_security_policy_from_version()`, and only set `cnofig->security_policy` after this FIPS mode check has happened.